rfcs/003-randomness.md

```diff
@@ -0,0 +1,151 @@
+
+## Existing Randomness Methods
+
+As in all blockchains, finding randomness in Polkadot is a complicated task;
+depending on the final use case, the randomness level may not be suited for
+the final application. This section will cover the randomness sources we have
+at our disposal.
+
+### Block Hash
+
+While deterministic — i.e. for two equal inputs (blocks), the output of the
+hashing function will always be the same — block hashes are not predictable,
+in other words, without running the hashing function, you cannot predict the
+final hash. While block hashes are not predictable, they are biasable by the
+block author, being possible to change small aspects of the block to influence
+the final hash, until a "more desirable" one is found.
+
+In Polkadot, Collators are responsible for block authoring, as such, they are
+the only ones that may alter the block to change the output hash. This change
+is not detectable by the validators because it keeps the block integrity,
+as such the final block (barring other changes) will be a valid block that
+eventually gets integrated in the chain.
+
+In the Polka Storage Parachain, the SPs are not Collators, but nothing stops
+a SP from running a malicious Collator node, meaning they can influence the
+value of a block for their own gain; however, collator selection has two
+features that play against this — invulnerables, which are collators we trust
+by default and the fact that multiple collators propose a block, making the
+determining the winning block a non-deterministic operation. As such, as long
+as there is at least one honest collator that is able to submit a winning block
+every so often, a dishonest SP will always be detected.
+
+Furthermore, as long as we use the latest block hash available (i.e. the
+parent block of the one being produced) even if the SP can bias the resulting
+hash, it will only provide them with an advantage of 6 seconds (block production
+time) + challenge lookback (10 minutes for the production testnet); additionaly
+this advantage does not stack, meaning that being 10 (or any arbitrary amount)
+minutes early to a proof does not enable the SP to start the next proof 10 (or
+any arbitrary amount) minutes early.
+
+### BABE Randomness
+
+As part of BABE consensus, VRFs are used to select the validator that will
+propose a block — the VRF [randomness is exposed in the relay chain proof][1],
+meaning we just need to "pull" that randomness into our parachain (easier
+said than done); this is the approach taken by Moonbeam ([MB1][2], [MB2][3]),
+and StorageHub ([SH1][4], [SH2][5]).
+
+This approach involves injecting the BABE randomness value into parachain
+storage, then, when necessary fetch it for usage. The main advantage is that
+this approach is verifiable, the main disadvantage is that the randomness
+takes a long time to refresh as it is dependent on BABE epochs (which last
+4 hours on mainnet).
```

Comment:
What are the consequences of using 4-hour randomness, how does verifiable help us in our case?